src/main/java/net/mcreator/ui/action/impl/workspace/SDKDownloader.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2021, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.action.impl.workspace;
+
+import net.mcreator.generator.GeneratorConfiguration;
+import net.mcreator.io.OS;
+import net.mcreator.io.UserFolderManager;
+import org.apache.commons.io.FileUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.net.URL;
+
+public class SDKDownloader {
+
+	private static final Logger LOG = LogManager.getLogger("Setup JDK");
+
+	public static void downloadJDK(GeneratorConfiguration generatorConfiguration) throws Exception {
+		String jdkVersion = generatorConfiguration.getJDKVersionOverride();
+		if (jdkVersion != null && !validateCustomJDK(generatorConfiguration)) {
+			LOG.info("Downloading JDK: " + jdkVersion);
+
+			String fileExtension;
+			if (OS.getOS() == OS.WINDOWS)
+				fileExtension = ".zip";
+			else
+				fileExtension = ".tar.gz";
+
+			FileUtils.copyURLToFile(

```

Comment:
I am not entirely sure how this can work, this downloads zip or tar and you then specify folder for the java home. You will need to extract the archive in the appropriate folder for this to work

Reply:
I tried several things for ProgressMonitorInputStream, but the dialog was never visible.

Reply:
I presume the UI then blocks when you download which is not ok. Also you need to add task to unzip the downloaded files